src/languages/handlebars.js

```diff
@@ -5,35 +5,57 @@ Author: Robin Ward <robin.ward@gmail.com>
 Description: Matcher for Handlebars as well as EmberJS additions.
 Category: template
 */
-
 function(hljs) {
   var BUILT_INS = {'builtin-name': 'each in with if else unless bindattr action collection debugger log outlet template unbound view yield'};
+  function MUSTACHE_CONTENTS() {
+    return  {
+      className: 'name',
+      begin: /[a-zA-Z\.-]+/,
+      keywords: BUILT_INS,
+      starts: {
+        endsWithParent: true, relevance: 0,
+        contains: [
+          hljs.QUOTE_STRING_MODE
+        ]
+      }
+    };
+  }
+
   return {
     aliases: ['hbs', 'html.hbs', 'html.handlebars'],
     case_insensitive: true,
     subLanguage: 'xml',
     contains: [
-    hljs.COMMENT('{{!(--)?', '(--)?}}'),
+      hljs.COMMENT('{{!(--)?', '(--)?}}'),
+      // raw block {{{{raw}}}} verbatim xml {{{{/raw}} {{handlebars}}
       {
         className: 'template-tag',
-        begin: /\{\{[#\/]/, end: /\}\}/,
-        contains: [
-          {
-            className: 'name',
-            begin: /[a-zA-Z\.-]+/,
-            keywords: BUILT_INS,
-            starts: {
-              endsWithParent: true, relevance: 0,
-              contains: [
-                hljs.QUOTE_STRING_MODE
-              ]
-            }
-          }
-        ]
+        begin: /\{\{\{\{\//, end: /\}\}\}\}/,
+        contains: [MUSTACHE_CONTENTS()]
+      },
+      {
+        className: 'template-tag',
+        begin: /\{\{\{\{/, end: /\}\}\}\}/,
+        contains: [MUSTACHE_CONTENTS()],
+        starts: {
+          end: '{{{{/',
+          returnEnd: true,
+          subLanguage: 'xml'
+        }
+     },
+      {
+        className: 'template-tag',
+        begin: /{{[#\/]/, end: /}}/,
+        contains: [MUSTACHE_CONTENTS()],
+      },
+      {
+        className: 'template-variable',
+        begin: '{{{', end: '}}}',
+        keywords: BUILT_INS
       },
       {
         className: 'template-variable',
-        begin: /\{\{/, end: /\}\}/,
+        begin: '{{', end: '}}',
```

Comment:
Reason this changed?

Reply:
I thought there was a different, possibly more efficient implementation for matching strings directly. Now, I have looked at the code and seen that everything is converted to regexp anyway. I will change this back to a regular expression.

I'm not sure escaping the curly brackets is really necessary. My IDE tells me that it isn't, but I'm not sure if this is true for every browser out there.

Reply:
Our consistency on // vs ‘’ is a bit all over the map but I just saw this changed but it didn’t really look like the actual content changed, so it jumped out at me.